app/models/hmpps_api/offender.rb

```diff
@@ -1,20 +1,289 @@
+    end
+
+    def ldu_name
+      ldu&.name
+    end
+
+    def ldu_email_address
+      ldu&.email_address
+    end
+
+    def delius_matched?
+      @case_information&.manual_entry == false
+    end
+
+    def allocated_com_name
+      @case_information&.com_name
+    end
+
+    def recalled?
+      @sentence.recall
+    end
+
+    def over_18?
+      age >= 18
+    end
+
+    def pom_responsible?
+      if @responsibility_override.nil?
+        HandoverDateService.handover(self).custody_responsible?
+      else
+        @responsibility_override.value == Responsibility::PRISON
+      end
+    end
+
+    def pom_supporting?
+      if @responsibility_override.nil?
+        HandoverDateService.handover(self).custody_supporting?
+      else
+        @responsibility_override.value == Responsibility::PROBATION
+      end
+    end
+
+    def com_responsible?
+      if @responsibility_override.nil?
+        HandoverDateService.handover(self).community_responsible?
+      else
+        @responsibility_override.value == Responsibility::PROBATION
+      end
+    end
+
+    def com_supporting?
+      if @responsibility_override.nil?
+        HandoverDateService.handover(self).community_supporting?
+      else
+        # Overrides to prison aren't actually possible in the UI
+        # If they were, we'd somehow need to decide whether COM is supporting or not involved
+        @responsibility_override.value == Responsibility::PRISON
+      end
+    end
+
+    def sentence_start_date
+      sentence.sentence_start_date
+    end
+
+    def full_name
+      "#{last_name}, #{first_name}".titleize
+    end
+
+    def full_name_ordered
+      "#{first_name} #{last_name}".titleize
+    end
+
+    def responsibility_override?
+      @responsibility_override.present?
+    end
+
+    # This list must only contain fields that are both supplied by
+    # https://api-dev.prison.service.justice.gov.uk/swagger-ui.html#//prisoners/getPrisonersOffenderNo
+    # and also by
+    # https://api-dev.prison.service.justice.gov.uk/swagger-ui.html#//locations/getOffendersAtLocationDescription
+    def initialize(api_payload, search_payload, category:, latest_temp_movement:, complexity_level:, booking_id:, prison_id:)
+      # It is expected that this method will be called by the subclass which
+      # will have been given a payload at the class level, and will call this
+      # method from it's own internal from_json
+      @first_name = api_payload.fetch('firstName')
+      @last_name = api_payload.fetch('lastName')
+      @offender_no = api_payload.fetch('offenderNo')
+      @convicted_status = api_payload['convictedStatus']
+      @date_of_birth = Date.parse(api_payload.fetch('dateOfBirth'))
+      @latest_temp_movement = latest_temp_movement
+      @cell_location = search_payload['cellLocation']
+      @complexity_level = complexity_level
+      @category = category
+
+      @booking_id = booking_id
+      @prison_id = prison_id
+    end
```

Comment:
I tend to prefer having the constructor method at the top of the class file – making it really obvious what goes 'in' to the object, before seeing the methods that come 'out' of it.

Right now it's just mixed in amongst all the other methods, which all happen to be in no particular order.

This could just be personal preference, so feel free to ignore.